lib/gem_content.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+class GemContent
+  CONTENT_ROOT_FORMAT = "gems/%{gem}/contents/"
+  CONTENT_KEY_FORMAT = "gems/%{gem}/contents/%{hash}"
+  HASHES_ROOT_FORMAT = "gems/%{gem}/checksums/"
+  HASHES_KEY_FORMAT = "gems/%{gem}/checksums/%{version}.sha256"
+  PATH_ROOT_FORMAT = "gems/%{gem}/paths/%{version}/"
+  PATH_KEY_FORMAT = "gems/%{gem}/paths/%{version}/%{path}"
+  SPEC_KEY_FORMAT = "gems/%{gem}/specs/%{gem}-%{version}.gemspec"
+
+  class << self
+    def fs
+      @fs || RubygemFs.instance
+    end
+
+    def mock!
+      @fs = RubygemFs.mock!
+    end
+  end
+
+  attr_reader :gem, :fs
+
+  def initialize(gem:, fs: self.class.fs)
+    raise ArgumentError, "gem must be a valid gem name" unless gem.try(:match?, Rubygem::NAME_PATTERN)
+    @gem = gem
+    @fs = fs
+  end
+
+  def get(hash)
+    return if hash.blank?
+    fs.get key(hash)
+  end
+
+  def keys
```

Comment:
does it maybe make sense to make this class enumerable and allow lazily enumerating keys?

Reply:
We could add this when it's needed. Should be easy to add.